datafusion/core/src/execution/context.rs

```diff
@@ -175,6 +175,26 @@ impl SessionContext {
         Self::with_config(SessionConfig::new())
     }
 
+    /// Finds any ListSchemaProviders and instructs them to reload tables from "disk"
+    pub async fn refresh_catalogs(&self) -> Result<()> {
```

Comment:
Moving this code to a common function on the context which we can use from `datafusion-cli`, tests, Ballista, etc.

Reply:
makes sense to me